src/Widget/SnippetWidget.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bolt\Widget;
+
+use Bolt\Snippet\Target;
+use Bolt\Snippet\Zone;
+
+class SnippetWidget extends BaseWidget
+{
+    protected $name = 'Nameless Snippet';
+    protected $type = 'snippet';
+    protected $target = Target::NOWHERE;
+    protected $zone = Zone::NOWHERE;
+    protected $priority = 200;
+    protected $snippet = '';
+
+    public function __construct(?string $snippet = null, ?string $name = null, ?string $target = null, ?string $zone = null)
+    {
+        if ($snippet !== null) {
```

Comment:
1. should they all be set as names?

2. no need to call setters here, types are already checked. then, those setters can be removed from BaseWidget as they are not used anymore.

Reply:
1. Nope, that was an oversight by me. Fixed.

2. partially updates. But, I disagree about removing the setters. We'll keep the setters, for when someone wants to use it as a fluent interface, or change a value after initialisation.